### PR TITLE
[NUI] Make Page Navigation APIs public

### PR DESCRIPTION
**Pull request by jaehyun0cho**
This is ACR patch to make Page Navigation APIs public.

This pull request includes properties and methods of Navigator and Page
classes.

NavigationPages are not going to be public not to make List<T> public
API.

To cover the usage of NavigationPages, the following APIs are added.
- public int PageCount
- public Page GetPage(int index)
- public int IndexOf(Page page)

### API Changes ###

 - ACR: TCSACR-417

### CODE REVIEW

**Comment by jaehyun0cho** [comment]
This PR requires https://github.com/Samsung/TizenFX/pull/2904

File: src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs [hunk 1]
```diff
@@ -426,7 +426,7 @@ public void Remove(Page page)
         /// </summary>
         /// <param name="index">The index of Navigator where a page will be removed.</param>
         /// <exception cref="ArgumentOutOfRangeException">Thrown when the index is less than 0, or greater than or equal to the number of pages.</exception>
-        [EditorBrowsable(EditorBrowsableState.Never)]
+        /// <since_tizen> 9 </since_tizen>
```
Comment (hyunjushin) [inline src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs:429]: if there isn't the page which has the index?

**Review by hyunjushin — APPROVED** [review]

File: src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs (hunk 1 above)
Comment (jaehyun0cho) [reply src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs:429]: Pushed pages are added to NavigationPages.
RemoveAt() checks if the index is valid by comparing 0 and NavigationPages.Count.
If the given index is not valid, then exception occurs.

**Review by Seoyeon2Kim — APPROVED** [review]
LGTM! Please update ACR number.

**Comment by jaehyun0cho** [comment]
> LGTM! Please update ACR number.
Thank you for your comment!

**Comment by jaehyun0cho** [comment]
ACR number is TCSACR-417

**Comment by dongsug-song** [comment]
@jaehyun0cho 
https://code.sec.samsung.net/jira/browse/TCSACR-417 
has not been accepted yet. 

and now I put the **\*\*NO MERGE\*\*** tag on this PR.
you can remove **\*\*NO MERGE\*\*** tag for yourself when this PR is ACR accepted and ready to be merged.
(**\*\*NO MERGE\*\*** tag has no restriction to submit PR or to edit this patch but just for notification and marking.
as you know, this PR can not be merged before the ACR is accepted.
thank you.

File: src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs [hunk 2]
```diff
@@ -340,18 +342,65 @@ public Page Pop()
             return curTop;
         }
 
+        /// <summary>
+        /// Returns the index of the given page in Navigator.
```
Comment (pkosko) [inline src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs:346]: I think that GetPage does not return the index, but Page object for the given index - please check.

File: src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs [hunk 3]
```diff
@@ -340,18 +342,65 @@ public Page Pop()
...
+        /// The indices of pages in Navigator are basically the order of pushing or inserting to Navigator.
+        /// So a page's index in Navigator can be changed whenever push/insert or pop/remove occurs.
+        /// </summary>
+        /// <param name="index">The index of a page in Navigator.</param>
```
Comment (pkosko) [inline src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs:350]: Probably 
```/// <returns> some descritpion```

should be added here.

**Review by pkosko — CHANGES_REQUESTED** [review]

File: src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs (hunk 2 above)
Comment (jaehyun0cho) [reply src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs]: Thank you for your comments.
I have fixed the summary.

File: src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs (hunk 3 above)
Comment (jaehyun0cho) [reply src/Tizen.NUI.Components/Controls/Navigation/Navigator.cs:363]: Thank you for your comments.
I have added the returns documentation.

**Review by pkosko — APPROVED** [review]
LGTM

**Pull request merged**